src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,40 @@
+interface Props {
+  title: string;
+  name: string;
+  value?: string;
+  label: string;
+  placeholder: string;
+  hasError?: boolean;
+  required?: boolean;
+  onChange: (newValue: string) => void;
+}
+
+export const TextField: React.FC<Props> = ({
+  title,
+  name,
+  value = title,
+  label,
+  placeholder,
+  hasError = false,
+  required = true,
+  onChange,
+}) => {
+  return (
+    <div className="field">
+      <label className="label" htmlFor="title">
+        {label}
+      </label>
+      <input
+        id={name}
+        type="text"
+        data-cy="titleInput"
+        placeholder={placeholder}
+        value={value}
+        onChange={event => onChange(event.target.value)}
```

Comment:
[CODE STYLE] - Don't use one-letter variable naming. In the 'onChange' handler, use a more descriptive name than 'event'.